ocs_ci/helpers/e2e_helpers.py

```diff
@@ -392,3 +410,335 @@ def validate_mcg_object_expiration(
+    ), "Failed: Default bs/bc are not in ready state"
+
+    # Create OBCs
+    logger.info("Creating OBCs after noobaa rebuild")
+    bucket_factory_session(amount=3, interface="OC", verify_health=True)
+
+
+def validate_noobaa_db_backup_recovery_locally_system(
+    self,
+    bucket_factory_session,
+    noobaa_db_backup_and_recovery_locally,
+    warps3,
+    mcg_obj_session,
+):
+    """
+    Test to verify Backup and Restore for Multicloud Object Gateway database locally
+    Backup procedure:
+    1. Create a test bucket and write some data
+    2. Backup noobaa secrets to local folder OR store it in secret objects
+    3. Backup the PostgreSQL database and save it to a local folder
+    4. For testing, write new data to show a little data loss between backup and restore
+    Restore procedure:
+    1. Stop MCG reconciliation
+    2. Stop the NooBaa Service before restoring the NooBaa DB. There will be no object service after this point
+    3. Verify that all NooBaa components (except NooBaa DB) have 0 replicas
+    4. Login to the NooBaa DB pod and cleanup potential database clients to nbcore
+    5. Restore DB from a local folder
+    6. Delete current noobaa secrets and restore them from a local folder OR secrets objects.
+    7. Restore MCG reconciliation
+    8. Start the NooBaa service
+    9. Restart the NooBaa DB pod
+    10. Check that the old data exists, but not s3://testloss/
+    Run multi client warp benchmarking to verify bug https://bugzilla.redhat.com/show_bug.cgi?id=2141035
+
+    """
+
+    # create a bucket for warp benchmarking
+    bucket_name = bucket_factory_session()[0].name
+
+    # Backup and restore noobaa db using fixture
+    noobaa_db_backup_and_recovery_locally(bucket_factory_session)
+
+    # Verify everything running fine
+    logger.info("Verifying all resources are Running and matches expected result")
+    self.sanity_helpers.health_check(tries=120)
+
+    # Run multi client warp benchmarking
+    warps3.run_benchmark(
+        bucket_name=bucket_name,
+        access_key=mcg_obj_session.access_key_id,
+        secret_key=mcg_obj_session.access_key,
+        duration="10m",
+        concurrent=10,
+        objects=100,
+        obj_size="1MiB",
+        validate=True,
+        timeout=4000,
+        multi_client=True,
+        tls=True,
+        debug=True,
+        insecure=True,
+    )
+
+    # make sure no errors in the noobaa pod logs
+    search_string = (
+        "AssertionError [ERR_ASSERTION]: _id must be unique. "
+        "found 2 rows with _id=undefined in table bucketstats"
+    )
+    nb_pods = get_noobaa_pods()
+    for pod in nb_pods:
+        pod_logs = get_pod_logs(pod_name=pod.name)
+        for line in pod_logs:
+            assert (
+                search_string not in line
+            ), f"[Error] {search_string} found in the noobaa pod logs"
+    logger.info(f"No {search_string} errors are found in the noobaa pod logs")
+
+
+class Run_fio_till_cluster_full:
```

Comment:
what is the use of class here?

Reply:
I kept the entire run_fio function and its cleanup function in this class to isolate from other code. So its easy to cleanup fio in the middle of test where I have requirement at https://github.com/red-hat-storage/ocs-ci/pull/10603/files#diff-983a0afd1203749ea2a488538ecc4b61dfe9531ca84d0574024b021d55e4476cR140

Please suggest me if there is any another way of implementing it.